### ISSUE TITLE
Docs: explain Dask concepts and Catalog representation better

### ISSUE THREAD

**Issue by hombit** [astronomy-commons/lsdb#386]
~~From `lincc-lsdb` slack channel, Gregory Dubois-Felsmann asked how do we get list of the columns for a catalog and highlighted that docs are not really helpful in discovering `Catalog.columns` attribute~~

It turns, that the real issue Gregory had is in understanding that `read_hipscat()` loads metadata only and what is actually shown when we run `display(catalog)` in Jupyter.

**Comment by nevencaplar** [comment]
It is basically this https://github.com/astronomy-commons/lsdb/issues/129 issue, that we closed... I guess more documentation improvements, but this would not help in this particular workflow that he described.

**Comment by gpdf** [comment]
It's a bit more complicated than that.  It was really a combination of the two, and a couple of other little details that add up to a workflow for someone who's new to a dataset being a little un-obvious -- and it isn't explored explicitly by the existing tutorials.

If you don't know what the columns are, you probably want to examine the schema of the dataset first.

A number of the current demos start with something like:

* `lsdb.read_hipscat(ztf_object_path, columns=["ps1_objid", "ra", "dec"])`

But a natural workflow for a newcomer to a dataset often is:

1. Find out what columns are available
2. Decide what columns are appropriate for an analysis job
3. Start setting up the query

So those people will want to do:

1. `zcat = lsdb.read_hipscat(ztf_object_path)`
2. Look at the columns available in `zcat` and make a decision
3. `zcat = lsdb.read_hipscat(ztf_object_path, columns=["ps1_objid", "ra", "dec"])`

There's something about the way the tutorials are organized that made it less than obvious to me that that's what I needed to do.

I wonder if a more natural workflow would replace 3. with something like a `zcat.limit_columns(["ps1_objid", "ra", "dec"])` -- perhaps you already have something like that available?

----

When it comes to finding out about the columns themselves, it wasn't all that easy to figure out about having to use both the `.columns` and `.dtypes` attributes, and how to use what comes back from `.dtypes`.  The return types from a lot of these operations are fairly obscure and multiply layered, so it's not easy to explore the reference documentation to figure this out in the absence of tutorials.  I did eventually get there after a fair amount of guessing and hitting '.' and `Tab` on various objects.

I found myself looking for some sort of `get_schema()` method that would have returned, perhaps, a list or dict of columns, with a dict with a number of attributes for each column (which might expand from just name and type to include descriptions, units, UCDs, etc. in the future).
